scripts/release.sh

```diff
@@ -0,0 +1,218 @@
+)
+
+    inserted=false
+    cat $changelog_path | while read ; do
+        line="$REPLY"
+        if ! $inserted && grep -q "^## $from_version\$" <<< $line ; then
+            inserted=true
+            echo "$changelog_next_version" >> $changelog_path.new
+            echo "" >> $changelog_path.new # formatting
+            echo "$line" >> $changelog_path.new
+        else
+            echo "$line" >> $changelog_path.new
+        fi
+    done
+
+    mv -f $changelog_path.new $changelog_path
+}
+
+color_echo green "" # formatting
+color_echo green "PRs since last release:"
+if [ -z "$prs_since_last_release" ]; then
+    color_echo yellow "No PRs found since last release"
+    color_echo yellow "Are you sure you want to make a release?"
+    read -p "Press Enter to continue..."
+else
+    color_echo green "$prs_since_last_release"
+fi
+color_echo green "" # formatting
+
+# prompt to update the app CHANGELOG
+app_changelog_path=CHANGELOG.md
+if `grep -q "^## $next_app_version\$" $app_changelog_path` ; then
+  color_echo yellow "WARN: App $app_changelog_path already contains an entry for version $next_app_version"
+  echo "Skipping $app_changelog_path update..."
+else
+    echo -n "Please update the App"
+    color_echo -n yellow " $app_changelog_path "
+    echo -n "with the changes for next version"
+    color_echo -n green " $next_app_version "
+    echo "" # formatting
+    echo "Save and close the file when complete"
+    read -p "Press Enter to continue..."
+
+    # add new changelog entry and open the file for editing
+    add_changelog_entry $app_changelog_path ngrok-operator $curr_app_version $next_app_version
+    $EDITOR $app_changelog_path
+fi
+echo "" # formatting
+echo "" # formatting
+
+# prompt to update the chart CHANGELOG
+chart_changelog_path=helm/ngrok-operator/CHANGELOG.md
+if `grep -q "^## $next_chart_version\$" $chart_changelog_path` ; then
+  color_echo yellow "WARN: Helm Chart $chart_changelog_path already contains an entry for version $next_chart_version"
+  echo "Skipping $chart_changelog_path update..."
+else
+    echo -n "Please update the Helm Chart"
+    color_echo -n yellow " $chart_changelog_path "
+    echo -n "with the changes for next version"
+    color_echo -n green " $next_chart_version "
+    echo "" # formatting
+    echo "Save and close the file when complete"
+    read -p "Press Enter to continue..."
+
+    # add new changelog entry and open the file for editing
+    add_changelog_entry $chart_changelog_path helm-chart $curr_chart_version $next_chart_version
+    $EDITOR $chart_changelog_path
+fi
+echo "" # formatting
+echo "" # formatting
+
+# files are updated, commit and share diff
+echo "Committing changes..."
+git add VERSION helm/ngrok-operator/Chart.yaml $app_changelog_path $chart_changelog_path
+git commit -m "Release ngrok-operator-$next_app_version helm-chart-$next_chart_version"
+echo "" # formatting
+echo "Please review the changes, then push the commit"
+read -p "Press Enter to continue..."
```

Comment:
nice, i was going to ask if there was any way to dry run and test this, but looks like you can just run until here and bail out correct?

Reply:
Yes essentially a dry-run because we only create a commit and do not push it. So the branch + commit is entirely throw-away-able.

Reply:
Oh and fwiw I did test this before opening the PR, here's a screenshot:

![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/kQZck6rnRcEsntiZQGPl/031a43db-b1fc-43b2-82f1-09d64b22f195.png)

![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/kQZck6rnRcEsntiZQGPl/d208ce94-6e67-40eb-bdef-f47d7ea4ba22.png)